pylinac/fieldparams.py

```diff
@@ -0,0 +1,615 @@
+        area_left = np.sum(values[:cax_idx + 1])
+        area_right = np.sum(values[cax_idx:])
+    symmetry = 100*abs(area_left - area_right)/(area_left + area_right)
+    return symmetry
+
+
+def deviation_diff(profile: SingleProfile, ifa: float = 0.8):
+    """Maximum deviation"""
+    if norm in ['max', 'max grounded']:
+        _, cax_val = profile.fwxm_center()
+    else:
+        _, cax_val = profile.profile_center()
+    try:
+        dmax = profile.field_calculation(field_width=ifa, calculation='max')
+        dmin = profile.field_calculation(field_width=ifa, calculation='min')
+    except ValueError:
+        raise ValueError("An error was encountered in the deviation calculation. The image is likely inverted. Try inverting the image before analysis with <instance>.image.invert().")
+    deviation = 100*(dmax - dmin)/cax_val
+    return deviation
+
+
+def deviation_max(profile: SingleProfile, ifa: float = 0.8):
+    """Maximum deviation"""
+    if norm in ['max', 'max grounded']:
+        _, cax_val = profile.fwxm_center()
+    else:
+        _, cax_val = profile.profile_center()
+    try:
+        dmax = profile.field_calculation(field_width=ifa, calculation='max')
+    except ValueError:
+        raise ValueError("An error was encountered in the deviation calculation. The image is likely inverted. Try inverting the image before analysis with <instance>.image.invert().")
+    deviation = 100*dmax/cax_val
+    return deviation
+
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Predefined Protocols - Do not change these. Instead copy a protocol, give it a new name, put it after these protocols
+# and add the protocol name to the dictionary PROTOCOLS.
+# ----------------------------------------------------------------------------------------------------------------------
+
+ALL = {
+    'Left edge (50%)': left_edge_50,
+    'Right edge (50%)': right_edge_50,
+    'Field size FWHM': field_size_50,
+    'Field size edge': field_size_edge_50,
+    'Field center edge': field_center_edge_50,
+    'Field center FWHM': field_center_fwhm,
+    'Penumbra 80-20% left': penumbra_left_80_20,
+    'Penumbra 80-20% right': penumbra_right_80_20,
+    'Flatness diff': flatness_dose_difference,
+    'Flatness ratio': flatness_dose_ratio,
+    'Symmetry diff': symmetry_point_difference,
+    'Symmetry ratio': symmetry_pdq_iec,
+    'Symmetry area': symmetry_area,
+    'Deviation max': deviation_max,
+    'Deviation diff': deviation_diff
+}
+
+VARIAN = {
+    'Left edge (50%)': left_edge_50,
+    'Right edge (50%)': right_edge_50,
+    'Field size': field_size_edge_50,
+    'Field center': field_center_edge_50,
+    'Penumbra 80-20% left': penumbra_left_80_20,
+    'Penumbra 80-20% right': penumbra_right_80_20,
+    'Flatness': flatness_dose_difference,
+    'Symmetry': symmetry_point_difference,
+}
+
+ORIGINAL_VARIAN = {
```

Comment:
Where does "original" varian come from?

Reply:
This returns the same values as the flatsym module. It is there for testing and backwards compatibility. As I mentioned I am finding small differences in the values, particularly for low res images (2D arrays) between pylinac and other programs that I have traced back to your method of interpolation and normalisation. This protocol uses your original functions. Perhaps it would be better to call it FLATSYM_VARIAN.

Reply:
Got it. Hmm, I'll have to think about the name.